lib/rules/no-unauthorized-global-properties.js

```diff
@@ -0,0 +1,279 @@
+"use strict";
+
+var standardGlobalProperties = [
+	"Array",
+	"ArrayBuffer",
+	"Boolean",
+	"DataView",
+	"Date",
+	"Error",
+	"EvalError",
+	"Float32Array",
+	"Float64Array",
+	"Function",
+	"Generator",
+	"GeneratorFunction",
+	"Infinity",
+	"Int16Array",
+	"Int32Array",
+	"Int8Array",
+	"InternalError",
+	"Intl",
+	"JSON",
+	"Map",
+	"Math",
+	"NaN",
+	"Number",
+	"Object",
+	"Promise",
+	"Proxy",
+	"RangeError",
+	"ReferenceError",
+	"Reflect",
+	"RegExp",
+	"Set",
+	"String",
+	"Symbol",
+	"SyntaxError",
+	"TypeError",
+	"URIError",
+	"Uint16Array",
+	"Uint32Array",
+	"Uint8Array",
+	"Uint8ClampedArray",
+	"WeakMap",
+	"WeakSet",
+	"WebAssembly",
+	"decodeURI",
+	"decodeURIComponent",
+	"encodeURI",
+	"encodeURIComponent",
+	"escape",
+	"eval",
+	"isFinite",
+	"isNaN",
+	"null",
+	"parseFloat",
+	"parseInt",
+	"undefined",
+	"unescape"
+];
+
+var nodeGlobalProperties = [
+	"__dirname",
+	"__filename",
+	"clearImmediate",
+	"clearInterval",
```

Comment:
intervals and timeout functions are node? Surely those are just standard (same with console)?